lib/NativeCall/Types.pm6

```diff
@@ -76,22 +78,24 @@ our class Pointer                               is repr('CPointer') {
             )
         }
     }
-    method ^parameterize(Mu:U \p, Mu:U \t) {
-        die "A typed pointer can only hold:\n" ~
-            "  (u)int8, (u)int16, (u)int32, (u)int64, (u)long, (u)longlong, num16, num32, (s)size_t, bool, Str\n" ~
-            "  and types with representation: CArray, CPointer, CStruct, CPPStruct and CUnion" ~
-            "not: {t.^name}"
-            unless t ~~ Int|Num|Bool || t === Str|void || t.REPR eq any <CStruct CUnion CPPStruct CPointer CArray>;
-        my $w := p.^mixin: TypedPointer[t.WHAT];
-        $w.^set_name: "{p.^name}[{t.^name}]";
-        $w;
+    method ^parameterize(Mu:U $this is raw, Mu:U $type is raw) {
+        when $type ~~ any(Int, Num, Str, void)
```

Comment:
I suspect the type check here and in `CArray.^parameterize` might make this fail some tests

Reply:
I think @MasterDuke17 's comments should be well-received. I'm already experiencing significant slow downs with the respect of compilation. I think I can wait a bit until @Kaiepi can rewrite this using if/elsif/else rather than when to improve performance.

Thanks much @Kaiepi and @MasterDuke17 !